reinstall-cli/src/config/mod.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{bail, ensure, Context, Result};
+use clap::Parser;
+use itertools::Itertools;
+use serde_json::Value;
+
+mod cli;
+
+#[derive(Debug)]
+pub(crate) struct ReinstallConfig {
+    pub(crate) bootc_image: String,
+}
+
+impl ReinstallConfig {
+    pub fn parse_from_cli(cli: cli::Cli) -> Result<Self> {
+        Ok(Self {
+            bootc_image: cli.bootc_image,
+        })
+    }
+
+    pub fn parse_from_config_file(config_bytes: &[u8]) -> Result<Self> {
+        let value: Value = serde_yaml::from_slice(config_bytes)?;
+
+        let mut value = value
+            .as_object()
+            .context("config file must be a YAML object")?
+            .clone();
+
+        let bootc_image = match value.remove("bootc_image") {
+            Some(value) => value
+                .as_str()
+                .context("bootc_image must be a string")?
+                .to_string(),
+            None => bail!("bootc_image is required"),
+        };
+
+        ensure!(
+            value.is_empty(),
+            "unknown keys {:?} in config file",
```

Comment:
Confused why we aren't using `#[derive(Deserialize)]` on the struct and https://serde.rs/container-attrs.html#deny_unknown_fields